### ISSUE TITLE
linter: not linting file when path specified contains $

### ISSUE THREAD

**Issue by bradleylandis** [oxc-project/oxc#15404]
### What version of Oxlint are you using?

1.26.0

### What command did you run?

oxlint --type-aware src/routes/posts.$id.tsx

### What does your `.oxlintrc.json` config file look like?

no config file

### What happened?

Finished in 2ms on 0 files with 103 rules using 10 threads.

**Comment by leaysgur** [comment]
🕵🏻 My guess, if you are using *ix OS (like macOS, Linux, etc), and seeing like this:

```sh
❯ oxlint --type-aware src/routes/posts.$id.tsx
Found 0 warnings and 0 errors.
Finished in 0ms on 0 files with 103 rules using 8 threads.
```

You should do like this instead.

```sh
oxlint --type-aware src/routes/posts.\$id.tsx
```

or

```sh
oxlint --type-aware 'src/routes/posts.$id.tsx'
```

Because `$` has special meanings.

---

If you try `oxlint --type-aware src/routes` and `posts.$id.tsx` is still not linted, please let us know.

**Comment by bradleylandis** [comment]
I'm on windows 11 and the way I encountered it was that my husky git hooks with lint-staged were passing when I had errors in the file.  I'll try your suggestions when manually running but I would assume that lint-staged would be taking care of passing the file name in properly.

**Comment by bradleylandis** [comment]
your first example above still said "0 files" but the second one did process the file.

**Comment by bradleylandis** [comment]
I also have a lint-staged rule for oxfmt and that does seem to process the file

**Comment by bradleylandis** [comment]
okay the issue was just that I need to pass --deny-warnings to oxlint when using it in the lint-staged husky hook
